### ISSUE TITLE
positionFixed + pagination does not work properly

### ISSUE THREAD

**Issue by gagarski** [ericgio/react-bootstrap-typeahead#431]
### Version

3.4.1

### Description

Pagination does not work properly when `positionFixed` or `bodyContainer` property is set to true.

### Steps to reproduce

1. Create the following component:
```jsx
<Typeahead options=["a", "b", "c", "d", "e"] maxResults={3} paginate positionFixed>
```
2. Click on this autocomplete and try to click the bottom option (load more results)
### Expected Behavior

More results are loaded

### Actual Behavior

Autocomplete menu is hidden. 

Further debugging shows that `onPaginate` callback is called if present.

Probably the same happens on selecting a regular value, but menu closing and calling `onSelect` is what the user wants in most cases.

My wild guess is that the reason is broken "click outside" handler which is called every time on menu item click when `positionFixed` is enabled

**Comment by ericgio** [comment]
Hey @gagarski! Thanks for the report and detailed repro steps. This should be fixed in v3.4.2. I don't love the solution, but it's somewhat temporary since v4.0 gets rid of the portal in favor of popper.js's `positionFixed` option.

**Comment by gagarski** [comment]
Awesome! Thanks for quick fix.
